src/ducks/Alerts/TypeSteps/NotificationsSettings/NotificationsSettings.js

```diff
@@ -0,0 +1,454 @@
+
+  const onWebhookChange = useCallback(
+    e => {
+      const channel = findWebHook(channels)
+
+      const val = e.target.value
+
+      if (channel) {
+        channel.webhook = val
+        handleFormValueChange({
+          field: 'channels',
+          value: channels
+        })
+        handleTitlesChange('channels', channels)
+      } else {
+        handleFormValueChange({
+          field: 'channels',
+          value: [
+            ...channels,
+            {
+              webhook: val
+            }
+          ]
+        })
+        handleTitlesChange('channels', [
+          ...channels,
+          {
+            webhook: val
+          }
+        ])
+      }
+
+      setWebhook(val)
+    },
+    [channels, handleFormValueChange]
+  )
+
+  const addOrRemove = useCallback(
+    (channel, flag) => {
+      const enabled = channels.indexOf(channel) !== -1
+
+      if (enabled && flag === true) {
+        return
+      }
+
+      if (!enabled && flag === false) {
+        return
+      }
+
+      let newChannels
+
+      if (enabled) {
+        newChannels = channels.filter(item => item !== channel)
+      } else {
+        newChannels = [...channels, channel]
+      }
+
+      handleFormValueChange({
+        field: 'channels',
+        value: newChannels
+      })
+      handleTitlesChange('channels', newChannels)
+    },
+    [channels, handleFormValueChange]
+  )
+
+  const toggleChannel = useCallback(
+    channel => {
+      switch (channel) {
+        case CHANNEL_NAMES.Webhook: {
+          let newChannels = []
+          const whChannel = findWebHook(channels)
+          if (!whChannel) {
+            newChannels = [
+              ...channels,
+              {
+                webhook
+              }
+            ]
+          } else {
+            newChannels = channels.filter(item => !isWebhookChannel(item))
+          }
+          handleFormValueChange({
+            field: 'channels',
+            value: newChannels
+          })
+          handleTitlesChange('channels', newChannels)
+          break
+        }
+
+        default: {
+          addOrRemove(channel)
+        }
+      }
+    },
+    [channels, handleFormValueChange, addOrRemove]
+  )
+
+  useEffect(() => {
+    if (!channels.length) {
+      return
+    }
+
+    if (!webhook) {
+      const whChannel = findWebHook(channels)
+      if (whChannel) {
+        setWebhook(whChannel.webhook)
+      }
+    }
+
+    const active = channels.filter(channel => !isDisabled(channel))
+
+    if (!channels.some(channel => active.indexOf(channel) !== -1)) {
+      handleFormValueChange({
+        field: 'channels',
+        value: active
+      })
+      handleTitlesChange('channels', active)
+    }
+  }, [channels])
+
+  useEffect(() => {
+    addOrRemove(CHANNEL_NAMES.Telegram, isTelegramConnected)
+  }, [isTelegramConnected])
+
+  useEffect(() => {
+    addOrRemove(CHANNEL_NAMES.Email, isEmailConnected)
+  }, [isEmailConnected])
```

Comment:
`addOrRemove` is not in the dependency list but is the `useCallback` closure function